wurst/closures/ClosureForGroups.wurst

```diff
@@ -56,6 +56,13 @@ public function forUnitsOfPlayer(player p, ForGroupCallback c)
 	GroupEnumUnitsOfPlayer(DUMMY_GROUP, p, filter)
 	popCallback()
 
+/** Executes the given closure for every existing unit */	
+public function forUnits(ForGroupCallback c)
+	pushCallback(c)
```

Comment:
you need to push/pop callbacks before each Enum call, not just before/after the loop.
Otherwise if there were another ForGroup closure inside the `forUnits`s one, it wouldn't work properly.